tools/clang_tidy.py

```diff
@@ -8,28 +8,35 @@
 import sys
```

Comment:
I didn't get to review this file when it was originally committed, so now you get some code review.

Can we have a comment at the top of this file explaining what this script does, and why you would prefer running it (as opposed to just running clang-tidy by yourself)?

Reply:
sure